### PR TITLE
Parse complex filters in URL (#465)

### PR DESCRIPTION
**Pull request by shinyichen**
This update added to the common URL parsing to handle complex filter parameters.

### CODE REVIEW

**Comment by shinyichen** [comment]
I made changes to recordEdit and viewer apps, so including @howdyjessie @jrchudy here.

**Comment by jrchudy** [comment]
Changes look good to me. I used a lot of this code in the parse function I wrote in ermrestJS and I'm sure there's some refactoring that can be done that we can look at. @howdyjessie do the changes for viewer look okay to you? I'm not sure if we need to be so precise, but they work.

File: viewer/viewer.app.js [hunk 1]
```diff
@@ -31,11 +31,11 @@
         // should we allow for improper URLs here?
         // what if there are 2 filters and the id filter is the second one.
         // Is that improper or should it be parsed and ignore the other filter?
-        angular.forEach(context.filters, function(filter) {
-            if(filter.name.toLowerCase() === "id") {
-                context.imageID = filter.value;
-            }
-        });
+        if (context.filter.type === "BinaryPredicate" &&
```
Comment (howdyjessie) [inline viewer/viewer.app.js]: Tiny nit here but I see `context.filter` accessed multiple times and it's slightly better to just store that in a variable for quicker access.

**Comment by howdyjessie** [comment]
Looks good to me too :) Merge at will.

**Pull request merged**